### DISCUSSION TITLE
Does this apply custom styling to Steam, or unstyle it?

### DISCUSSION THREAD

**Discussion by beedell-roke** [tkashkin/Adwaita-for-Steam#253] [Q&A] [answered]
I want to know whether this applies a theme to custom-drawn controls, which imitates GTK4's Adwaita, or whether it unstyles Steam's GTK controls (since Steam relies upon GTK). I ask because I use KDE Plasma 6, meaning that if it *unstyles* the controls, when (or if) [`kde-gtk-config`](https://wiki.archlinux.org/index.php?title=Uniform_look_for_Qt_and_GTK_applications&oldid=807141#kde-gtk-config_%22System_Settings_%3E_Application_Style_%3E_GTK%22_menu_missing:~:text=bindings%20for%20GTK.-,kde%2Dgtk%2Dconfig,-%22System%20Settings%20%3E) gains GTK4 support, I'll be able to force Steam to adhere to my system theme, Breeze.

**Comment by tkashkin** [comment]
Steam's UI is built on CEF, it does not use GTK. Each window or popup is a separate webview. This skin is just a bunch of CSS files that imitate libadwaita styles and are loaded on top of Steam's default CSS.

> I ask because I use KDE Plasma 6, meaning that if it unstyles the controls, when (or if) [kde-gtk-config](https://wiki.archlinux.org/index.php?title=Uniform_look_for_Qt_and_GTK_applications&oldid=807141#kde-gtk-config_%22System_Settings_%3E_Application_Style_%3E_GTK%22_menu_missing:~:text=bindings%20for%20GTK.-,kde%2Dgtk%2Dconfig,-%22System%20Settings%20%3E) gains GTK4 support, I'll be able to force Steam to adhere to my system theme, Breeze.

You can apply some of the predefined [color themes](https://github.com/tkashkin/Adwaita-for-Steam/tree/master/adwaita/colorthemes) or make your own. You can also apply [custom CSS](https://github.com/tkashkin/Adwaita-for-Steam/tree/master/custom) and change some of the [CSS variables](https://github.com/tkashkin/Adwaita-for-Steam/tree/master/adwaita/css/_root).

**Reply by beedell-roke** [reply]
> https://github.com/tkashkin/Adwaita-for-Steam/discussions/253#discussioncomment-9653737

@tkashkin, apologies for the basis of my question being incorrect. Considering that in the terminal, Steam outputs GTK warnings, and some statements online that it uses it, I considered that it must be fact.

However, I was referring solely to its surrounding GUI, like the top bar and configuration window, rather than the evidently embedded browser engine (which functions as its primary GUI). Though if by CEF, you mean https://github.com/cztomczak/cef2go/blob/a8da6e1377d1b9ec0087cc7bfe8fa8a11d2c3db7/README.md#introduction:~:text=CEF2go%20is%20an%20open%20source%20project%20founded%20by%20Czarek%20Tomczak%20in%202014%20to%20provide%20Go%20bindings%20for%20the%20Chromium%20Embedded%20Framework%20(CEF).%20CEF2go%20can%20act%20as%20a%20GUI%20toolkit%2C%20allowing%20you%20to%20create%20an%20HTML%205%20based%20GUI%20in%20your%20application, then I suppose that even the surrounding elements would be web technology-based regardless. Thanks for the response.

**Reply by tkashkin** [reply, marked as answer]
@RokeJulianLockhart CEF is [Chromium Embedded Framework](https://github.com/chromiumembedded/cef). The whole Steam UI, including window controls, Big Picture, and overlay are now entirely web-based and are rendered using CEF.

Previously, before the last big UI update, Steam used Valve's own UI framework called VGUI and only used CEF to embed web pages like Store and Community. I think Steam used VGUI since the beginning. VGUI was also used in most of Valve's games from Half-Life up to earlier versions of CS:GO.

Steam never used GTK directly for UI, but it or some of its dependencies probably use GLib or some other GTK-adjacent components internally.
